src/language_concepts/data_types.md

```diff
@@ -1,206 +1,192 @@
-# Data types
+# Data Types
 
-Noir has several varieties of data types: the primitive `Field` type, integer types, `bool`, arrays, structs, and tuple types.
+Every value in Noir is specified with a certain data type, which informs Noir tools how to work with that data when processing Noir programs.
 
-Although each value in a constraint system is fundamentally a field element, we add a layer of abstraction over this;
-each value can be _concealed_ (a private type) or _revealed_ (a public type).
+All values in Noir are fundamentally `Field` elements. For a more approachable developing experience, abstractions are added on top to introduce different data types in Noir.
 
-A **private value** is known only to the Prover, while, a **public value** is known by the Prover and Verifier.
-All public and private values are primitive types, though individual
-fields of a struct may also be marked private or public.
+Noir has two category of data types: primitive types (e.g. `Field`, integer, `bool`) and compound types (e.g. array, tuple, struct). Each value can either be private or public.
 
-Note that the terms private and public when applied to a type (e.g. in `pub Field`) have a different meaning than when
-applied to a function `pub fn foo() { ... }`.
-The later is a visibility modifier familiar to those coming from existing
-langauges, while the former can be thought of as a visibility modifier for the Verifier.
-So a `pub Field` would be visible to the Verifier while a `Field` would not (private is the default modifier).
+## Private & Public Types
 
-## Primitive Types
-
-Any of these types can be private (the default) or public (via a `pub` modifier).
-
-Note that private types are also referred to as witnesses.
+A **private value** is known only to the Prover, while a **public value** is known by both the Prover and Verifier. All public and private values are primitive types, though individual fields of compound types may also be marked private or public.
```

Comment:
We don't allow marking something explicitly private since it is the default. It'd be more accurate to say fields are also private by default but can be explicitly marked public.